proto/streamlit/proto/NumberInput.proto

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2018-2019 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+syntax = "proto3";
+
+
+
+message NumberInput {
+  string id = 1;
+  string label = 2;
+  float default = 3;
+
+  oneof optional_min {
```

Comment:
Remove `oneof`.

In proto3, all fields are actually optional. And there's [a reason](https://stackoverflow.com/a/31814967) for that.

In this case, you're actually not trying to make something optional but instead trying to allow it to be set to `null`. But the meaning of `null` is not clear, and so it should be avoided. In this specific case, `null` means `- Infinity`. So why not just fill this value with that? `-Infinity` is a valid floating point number, so you can just do this in Python:

```python
number_input.min = float("-inf")
```

and read if back like this in JS:

```javascript
if (element.get('min') == -Infinity) {
  ...
}
```

Reply:
Done